src/metabase/api/search.clj

```diff
@@ -192,6 +144,40 @@
     (sql.helpers/where query [:= id :database_id])
     query))
 
+(mu/defn ^:private replace-select :- :map
```

Comment:
I have another PR that uses this mechanism to improve search perm https://github.com/metabase/metabase/pull/34353